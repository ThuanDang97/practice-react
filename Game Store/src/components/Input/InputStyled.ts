import { styled, TextField } from '@mui/material'

// Types
import { IInputProps } from '@components/Input'

export const InputStyled = styled(TextField)<IInputProps>`
  background-color: transparent;
  margin-bottom: 10px;

  &.MuiTextField-root {
    position: relative;
    width: 100%;
    ${({ mbottom }) => `margin-bottom: ${mbottom}px`};
  }

  &::after {
    position: absolute;
    top: 32px;

    display: ${({ displayline }) => displayline || 'block'};
    content: '';
    width: 15%;
    min-width: 60px;
    border-top: 4px solid
      ${({ error, theme }) =>
        (error && theme.palette.error.dark) || theme.palette.primary.main};
    border-right: 4px solid transparent;
  }
`
