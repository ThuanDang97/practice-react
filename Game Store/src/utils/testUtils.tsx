import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import { BrowserRouter as Router } from 'react-router-dom'

// Themes
import theme from '@self-types/Themes'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <AuthProvider>
        <CartProvider>
          <Router>{component}</Router>
        </CartProvider>
      </AuthProvider>
    </ThemeProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
