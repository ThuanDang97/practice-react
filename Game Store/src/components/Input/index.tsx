import { InputProps } from '@mui/material'

// Styles
import { InputStyled } from '@components/Input/InputStyled'

export interface IInputProps extends InputProps {
  displayline?: 'none' | 'block'
  mbottom?: number
}

const Input = ({
  error,
  onChange,
  title,
  displayline,
  placeholder,
  value,
  type,
  mbottom,
  name,
}: IInputProps) => {
  return (
    <InputStyled
      label={title}
      color="primary"
      focused
      placeholder={placeholder}
      variant="standard"
      error={error}
      onChange={onChange}
      displayline={displayline}
      value={value}
      type={type}
      mbottom={mbottom}
      name={name}
    />
  )
}

export default Input
