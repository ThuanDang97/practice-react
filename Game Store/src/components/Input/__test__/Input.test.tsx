import { cleanup, fireEvent, screen } from '@testing-library/react'

// Utils
import { render } from '@utils/testUtils'

// Components
import Input from '@components/Input'

describe('Input component', () => {
  const handleOnChange = jest.fn()

  afterEach(() => {
    cleanup()
  })

  it('should render Input', () => {
    const { container } = render(
      <Input placeholder="Enter here" onChange={handleOnChange} />,
    )

    expect(container).toMatchSnapshot()
  })

  it('should call onChange Input ', () => {
    render(<Input placeholder="Enter here" onChange={handleOnChange} />)

    const input = screen.getByPlaceholderText('Enter here') as HTMLInputElement

    fireEvent.change(input, {
      target: { value: 'Input Value' },
    })

    expect(handleOnChange).toBeCalled()
    expect(input).toHaveValue('Input Value')
  })
})
