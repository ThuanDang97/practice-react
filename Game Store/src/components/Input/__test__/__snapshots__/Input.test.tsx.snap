// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render Input 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-1pcdsgt-MuiFormControl-root-MuiTextField-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=":r0:"
        placeholder="Enter here"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
